pkg/test/framework/components/echo/calloptions.go

```diff
@@ -24,6 +24,55 @@ import (
 	"istio.io/istio/pkg/test/echo/common/scheme"
 )
 
+// HTTP settings
+type HTTP struct {
+	// If true, h2c will be used in HTTP requests
+	HTTP2 bool
+
+	// If true, HTTP/3 request over QUIC will be used.
+	// It is mandatory to specify TLS settings
+	HTTP3 bool
+
+	// Path specifies the URL path for the HTTP(s) request.
+	Path string
+
+	// Method to send. Defaults to HTTP.
+	Method string
+
+	// Headers indicates headers that should be sent in the request. Ignored for WebSocket calls.
+	// If no Host header is provided, a default will be chosen for the target service endpoint.
+	Headers http.Header
+
+	// FollowRedirects will instruct the call to follow 301 redirects. Otherwise, the original 301 response
+	// is returned directly.
+	FollowRedirects bool
+
+	// HTTProxy used for making ingress echo call via proxy
+	HTTPProxy string
+}
+
+// TLS settings
+type TLS struct {
+	// Use the custom certificate to make the call. This is mostly used to make mTLS request directly
+	// (without proxy) from naked client to test certificates issued by custom CA instead of the Istio self-signed CA.
+	Cert, Key, CaCert string
+
+	// Use the custom certificates file to make the call.
+	CertFile, KeyFile, CaCertFile string
+
+	// Skip verify peer's certificate.
+	InsecureSkipVerify bool
+
+	Alpn       []string
+	ServerName string
+}
+
+// TCP settings
+type TCP struct {
+	// ExpectedResponse asserts this is in the response for TCP requests.
+	ExpectedResponse *wrappers.StringValue
+}
+
 // CallOptions defines options for calling a Endpoint.
 type CallOptions struct {
 	// Target instance of the call. Required.
```

Comment:
we may want to make port a non-pointer btw. That has been a minor inconvenience for a while